### PR TITLE
Proposal of tests refactorization

### PR DESCRIPTION
**Pull request by gponsu**

### CODE REVIEW

File: test/controllers/api/registrations_controller_test.rb [hunk 1]
```diff
@@ -2,33 +2,25 @@
...
+    @invitation = Invitation.create(email: 'johndoe@example.com')
+    @user_data = { email: @invitation.email, password: 'my-password' }
   end
 
   test 'create from invitation success' do
-    invitation = Invitation.first
-    user_data = { email: invitation.email, password: 'my-password' }
-    user_number = User.count
-    get :create, email: invitation.email, token: invitation.token, user_data: user_data
+    assert_difference 'User.count' do
```
Comment (d2bit) [inline test/controllers/api/registrations_controller_test.rb:15]: Set difference **1**
`assert_difference 'User.count' , 1 do`

Comment (gponsu) [inline test/controllers/api/registrations_controller_test.rb:15]: 1 is default :)

File: test/controllers/api/registrations_controller_test.rb [hunk 2]
```diff
@@ -2,33 +2,25 @@
 
 class Api::RegistrationsControllerTest < ActionController::TestCase
   def setup
-    Invitation.create(email: 'johndoe@example.com')
-  end
-
-  def teardown
```
Comment (d2bit) [inline test/controllers/api/registrations_controller_test.rb]: Why do you remove the `teardown`? I don't like keeping the database dirty

File: test/controllers/api/registrations_controller_test.rb (hunk 1 above)
Comment (d2bit) [inline test/controllers/api/registrations_controller_test.rb:15]: true! :sweat_smile: 
`assert_difference(expression, difference = 1, message = nil, &block)`

File: test/controllers/api/sessions_controller_test.rb [hunk 3]
```diff
@@ -14,33 +14,29 @@ def teardown
     get :create, email: user.email, password: 'secret'
 
     assert_response :success
-    assert_not_nil user.reload.auth_token
+    assert_not_empty user.reload.auth_token
```
Comment (d2bit) [inline test/controllers/api/sessions_controller_test.rb:17]: better `assert_not_nil`

**Pull request closed without merging**